### ISSUE TITLE
`ioctl` function signature is wrong

### ISSUE THREAD

**Issue by mbr** [dcuddeback/ioctl-rs#3]
In C:

```c
       #include <sys/ioctl.h>

       int ioctl(int d, unsigned long request, ...);
```

Library (at least inside `os::linux`):

```
extern "C" {
    pub fn ioctl(fd: c_int, request: c_int, ...) -> c_int;
}
```

I believe `request` should be a `c_uint`, otherwise ioctls >= `0x8000_0000` or higher will cause overflow warnings.

**Comment by dcuddeback** [comment]
@mbr Good catch. It should be `c_ulong` on x86_64. `c_int` is still correct for ARM, though. The manpage on my Beaglebone Black says:

```c
int ioctl(int d, int request, ...);
```

**Comment by dcuddeback** [comment]
On the other hand, `tty_ioctl(4)` says `int ioctl(int fd, int cmd, ...)` on x86_64. I'm not sure what to make of the discrepancy. It's worth further investigation.

**Comment by mbr** [comment]
FWIW, I've patched it and have successfully ran code on the BeagleBone
using unsigned ints.

**Comment by dcuddeback** [comment]
@mbr I checked against the header files. Looks like `unsigned int` is correct on x86_64 and ARM. Just pushed a fix for it and published it as v0.2.0. I bumped from v0.1.x since changing the type would break compilation of downstream projects.
